lib/stsci/tools/bitmask.py

```diff
@@ -8,18 +9,399 @@
 """
 
 import numpy as np
+import warnings
+import six
+from astropy.utils import deprecated
 
-__version__ = '0.2.0'
-__vdate__ = '23-February-2017'
+
+__version__ = '1.0.0'
+__vdate__ = '16-March-2017'
 __author__ = 'Mihai Cara'
 
+
+__all__ = ['interpret_bits_value', 'interpret_bit_flags', 'bitmask2mask',
+           'bitfield_to_boolean_mask', 'is_bit_flag']
+
+
 # Revision history:
 # 0.1.0 (29-March-2015) - initial release based on code from stsci.skypac
 # 0.1.1 (21-February-2017) - documentation typo fix
 # 0.2.0 (23-February-2017) - performance and stability improvements. Changed
 #       default output mask type from numpy.uint8 to numpy.bool_.
+# 1.0.0 (16-March-2017) - Multiple enhancements:
+#       1. Deprecated 'interpret_bits_value()'in favor of
+#          'interpret_bit_flags()' which now takes 'flip_bits' argument to flip
+#          bits in (list of) integer flags.
+#       2. Deprecated 'bitmask2mask()' in favor of 'bitfield_to_boolean_mask()'
+#          which now also takes 'flip_bits' argument.
+#       3. Renamed arguments of 'interpret_bit_flags()' and
+#          'bitfield_to_boolean_mask()' to be more technically correct.
+#       4. 'interpret_bit_flags()' and 'bitfield_to_boolean_mask()' now
+#          accept Python lists of bit flags (in addition to integer bitmasks
+#          and string comma- (or '+') separated lists of bit flags).
+#       5. Added 'is_bit_flag()' function to check if an integer number has
+#          only one bit set (i.e., that it is a power of 2).
+
+
+def is_bit_flag(n):
+    """
+    Verifies if the input number is a bit flag (i.e., an integer number that is
+    an integer power of 2).
+
+    Parameters
+    ----------
+    n : int
+        A positive integer number. Non-positive integers are considered not to
+        be "flags".
+
+    Returns
+    -------
+    bool
+        ``True`` if input ``n`` is a bit flag and ``False`` if it is not.
+
+    """
+    if hasattr(n, '__iter__'):
+        return all(map(is_bit_flag, n))
+
+    if n < 1:
+        return False
+
+    if n < 512:
+        s = 1
+        while True:
+            ns = n >> s
+            if ns == 0:
+                return True
+            if ns << s != n:
+                return False
+            s += 1
+
+    # more efficient for larger values of 'n':
+    return n == (1 << int(np.log2(n)))
```

Comment:
Complimenting what @MSeifert04 said, `np.log2` is a bit slower than `math.log2`.

```python
In [19]: %timeit math.log2(1 << 32)
10000000 loops, best of 3: 124 ns per loop

In [20]: %timeit math.log2(1 << 64)
10000000 loops, best of 3: 126 ns per loop
```

```python
In [21]: %timeit np.log2(1 << 32)
1000000 loops, best of 3: 998 ns per loop

In [22]: %timeit np.log2(1 << 63)
1000000 loops, best of 3: 1.19 µs per loop
```

I assume `np.log2` uses a signed 64-bit integer type under the hood. When you shift left 64 bits the function crashes. ;)

```python
In [23]: np.log2(1 << 64)
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-33-d1c17f7c05af> in <module>()
----> 1 np.log2(1 << 64)

AttributeError: 'int' object has no attribute 'log2'
```